addons/survey/survey.py

```diff
@@ -746,7 +758,12 @@ def validate_datetime(self, cr, uid, question, post, answer_tag, context=None):
             # Answer is not in the right range
             try:
                 dateanswer = datetime.datetime.strptime(answer, DF)
-                if not (datetime.datetime.strptime(question.validation_min_date, DF) <= dateanswer <= datetime.datetime.strptime(question.validation_max_date, DF)):
+                min_date = question.validation_min_date and datetime.datetime.strptime(question.validation_min_date, DF) or False
+                max_date = question.validation_max_date and datetime.datetime.strptime(question.validation_max_date, DF) or False
+
+                if (min_date and max_date and not(min_date <= dateanswer <= max_date)) or \
+                    (min_date and not(min_date <= dateanswer)) or \
+                    (max_date and not(dateanswer <= max_date)):
```

Comment:
@rim-odoo  i refactor my code.

But the code for.,
In "Date and Time" Question Type, if validate entry True then Minimum and Maximum Date are show,and this both are not required field.
Then if we enter any of one then validate function has error(must be string, not bool).
so here i check if any of one enter then check 'input date' according to it. 
For ex. if we enter only Minimum Date then 'input date' check on min date(min_date <= dateanswer).
![survey_datetime](https://cloud.githubusercontent.com/assets/7643769/6388828/ffb5af88-bdc3-11e4-852f-a6cef7af5386.png)